### PR TITLE
1861 improve remesh test

### LINKED ISSUE #1861: improve remesh test
various improvements needed: 

- [x] check 2D 3D required entities
- [x] parallel checks
- [ ] ale checks

### LINKED ISSUE #1863: Required facets may be missing
Requiring facets to be preserved may fail.
This is probably due to the hack that allows to keep track of mesh relationships

- [ ] keep relation 
- [x] no relation but required elements
- [x] no required elements

### LINKED ISSUE #1864: Allow parametrization of remeshing tool
Many options are available from mmg and parmmg.
They should be available via json and command line option

### PR DESCRIPTION
**Pull request by prudhomm**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?
- [x] Have you written Doxygen comments in your contribution ?

-----
- closes #1861 
- closes #1863
- closes #1864

### CODE REVIEW

**Comment by prudhomm** [comment]
There are still a few changes missing.

File: feelpp/feel/feelmesh/filters.hpp [hunk 1]
```diff
@@ -53,6 +78,44 @@ make_elements_wrapper()
...
+
+
+//!
+//! @return the worldcomm associated to the mesh stored in the range
+//! @ingroup Mesh
+//!
+template <typename RangeT, std::enable_if_t<is_range_v<RangeT>, int> = 0>
+WorldComm const& worldComm( RangeT const& range )
+{
+    return boost::unwrap_ref( *begin(range) ).mesh()->worldComm();
```
Comment (vincentchabannes) [inline feelpp/feel/feelmesh/filters.hpp:101]: will crash if the range is empty on one process. Maybe return the shared pointer if non null else a null pointer

File: feelpp/feel/feelmesh/filters.hpp [hunk 2]
```diff
@@ -53,6 +78,44 @@ make_elements_wrapper()
...
+ * @tparam S 
+ * @tparam ITERATOR 
+ * @tparam CONTAINER 
+ * @param range 
+ * @return int the rank of the entity range
+ */
+template <typename RangeT,std::enable_if_t<is_range_v<RangeT>,int> = 0>
+int rank( RangeT const& range )
+{
+    return worldComm( range ).globalRank();
```
Comment (vincentchabannes) [inline feelpp/feel/feelmesh/filters.hpp:116]: idem here, if empty return invalid_v<rank_type>. Replace int by rank_type

File: feelpp/feel/feelmesh/filters.hpp (hunk 1 above)
Comment (vincentchabannes) [reply feelpp/feel/feelmesh/filters.hpp:101]: probably the new struct that you develop for the range will fix that

**Review by vincentchabannes — APPROVED** [review]

File: feelpp/feel/feelmesh/filters.hpp (hunk 1 above)
Comment (prudhomm) [reply feelpp/feel/feelmesh/filters.hpp:101]: oui

**Pull request merged**